lib/minify.js

```diff
@@ -11,12 +11,25 @@ module.exports = function(uglify, log) {
       opts = {};
     }
 
-    return defaultsDeep(
+    //The nameCache option should be a mutable reference. Do not clone it.
+	var nameCache;
+    if (opts.nameCache !== undefined) {
+    	nameCache = opts.nameCache;
+    	delete opts.nameCache;
+    }
+
+    var defaults = defaultsDeep(
```

Comment:
If you manually set `nameCache` on the first argument to `defaultsDeep` from the provided `opts` object, it will be preserved.